### PR TITLE
make rspec and yard inclusion optional and do not install in production/stage

### PR DESCRIPTION
**Pull request by drh-stanford**
The goal of this PR is to exclude the `test` and `development` gems from our stage/prod deployments. That involves making rspec and yard optional in the Rakefile. I wasn't able to test this because deploys are broken on master right now.

Connects to #18

### CODE REVIEW

**Comment by ndushay** [comment]
add deploy to the gem groups we don't install?

File: config/deploy/production.rb [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 Capistrano::OneTimeKey.generate_one_time_key!
 
+set :bundle_without, %w(test development).join(' ')
```
Comment (ndushay) [inline config/deploy/production.rb]: add deploy?

File: config/deploy/stage.rb [hunk 2]
```diff
@@ -2,4 +2,5 @@
 
 Capistrano::OneTimeKey.generate_one_time_key!
 
+set :bundle_without, %w(test development).join(' ')
```
Comment (ndushay) [inline config/deploy/stage.rb]: add deploy?

**Comment by ndushay** [comment]
curiously, deploying to dev I got:
```
00:25 bundler:install
      01 /usr/local/rvm/bin/rvm default do bundle install --path /opt/app/was/was-registrar/shared/bundle --without development test --deployment --quiet
```

**Comment by ndushay** [comment]
Another thing to clean up:

```
03:22 passenger:restart
      01 passenger-config restart-app /opt/app/was/was-registrar --ignore-app-not-running
      01 There are no Phusion Passenger-served applications running whose paths begin with '/opt/app/was/was-registrar'.
      01
```

**Comment by drh-stanford** [comment]
The passenger error might be a puppet thing. I'll take a look.

Re: `deploy` I think `--deployment` will skip the `deploy` gem group from install. I looked at SearchWorks and EarthWorks and neither of them have `deploy` in the `bundle_without` clause.

https://github.com/sul-dlss/SearchWorks/blob/master/config/deploy/production.rb#L3
https://github.com/sul-dlss/earthworks/blob/master/config/deploy/production.rb#L4

**Comment by drh-stanford** [comment]
Re: passenger, it seems to be working now. the passenger processes weren't running so that might have been the problem.

```
01:27 passenger:restart
      01 passenger-config restart-app /opt/app/was/was-registrar --ignore-app-not-running
      01 Restarting /opt/app/was/was-registrar/current (development)
    ✔ 01 was@was-registrar-dev.stanford.edu 1.534s
```

**Comment by ndushay** [comment]
so it looked like cap  deployed without development or test  by default, because I deployed from master.  Am I wrong?

**Comment by drh-stanford** [comment]
oh yeah, that is curious. I looked at other projects and they're using the `bundle_without` explicitly:

https://github.com/sul-dlss/argo/blob/master/config/deploy/stage-a.rb#L5

but indeed that is the default for `bundle_without`. See https://github.com/capistrano/bundler

**Comment by drh-stanford** [comment]
I've removed the `bundle_without` statements.

**Comment by drh-stanford** [comment]
Looks to work now:

```
[was@was-registrar-dev current]$ cat .bundle/config 
---
BUNDLE_FROZEN: '1'
BUNDLE_PATH: "/opt/app/was/was-registrar/shared/bundle"
BUNDLE_WITHOUT: development:test
BUNDLE_DISABLE_SHARED_GEMS: '1'
[was@was-registrar-dev current]$ bundle exec rake spec
Please install the rspec gem to run tests.
[was@was-registrar-dev current]$ bundle exec rake doc
mkdir -p doc
Please install the YARD gem to generate rdoc.
[was@was-registrar-dev current]$ bundle exec rake routes
      Prefix Verb     URI Pattern                            Controller#Action
        root GET      /                                      seeds#index
crawls_index GET      /crawls/index(.:format)                crawls#index
             GET|POST /:controller(/:action(/:id))(.:format) :controller#:action
```

**Comment by drh-stanford** [comment]
Re: `deployment` in bundle without. I did find an example of a sul-dlss configuration that included `deployment` to exclude. See https://github.com/sul-dlss/revs-indexer-service/tree/88328268aeef8144fa6ecf4a7a095e274578fdf3/config/deploy

As far as I can tell, we don't have a standard practice on this, and adding `deployment` is indeed the correct thing to do. So I've added a commit for it (8c4749b)

File: Gemfile [hunk 3]
```diff
@@ -18,17 +18,25 @@ gem 'sdoc', '~> 0.4.0',          group: :doc
 gem 'spring',        group: :development
 
 gem 'nokogiri'
-gem "rspec"
-gem 'rspec-rails'
+
 gem 'rest-client'
-gem 'equivalent-xml'
 gem 'coveralls', require: false
```
Comment (ndushay) [inline Gemfile]: this is for testing also.

**Comment by ndushay** [comment]
Looks good now except coveralls should be in the test group.  I was just getting to the same research area on capistrano/bundler

**Comment by drh-stanford** [comment]
I've moved `coveralls` into the `test` group. There's also `sdoc` -- dunno why it's included. Maybe that goes in `development`?

**Comment by ndushay** [comment]
yay!

**Pull request merged**